src/Pulsar.Client/Internal/ConsumerImpl.fs

```diff
@@ -412,8 +444,9 @@ type ConsumerImpl internal (consumerConfig: ConsumerConfiguration, clientConfig:
                             let messagesFromQueue = removeExpiredMessagesFromQueue(messageIds);
                             let batches = messageIds |> Seq.chunkBySize MAX_REDELIVER_UNACKNOWLEDGED
                             for batch in batches do
+                                let batch' = batch |> Array.filter (fun messageId -> processDeadLetters messageId |> not)
```

Comment:
can you please rename batch' to nonDeadBatch (if I understand correctly)

Reply:
Done.